lms/static/scripts/frontend_apps/components/dashboard/test/DashboardActivityFilters-test.js

```diff
@@ -484,6 +495,161 @@ describe('DashboardActivityFilters', () => {
+    });
+
+    it('invokes onChange callback', () => {
+      const wrapper = createComponentWithSegments();
+      const select = getSegmentsSelect(wrapper);
+
+      select.props().onChange(['foo', 'bar']);
+
+      assert.calledWith(onSegmentsChange, ['foo', 'bar']);
+    });
+
+    ['groups', 'sections'].forEach(type => {
+      it('sets label based on segment type', () => {
+        const wrapper = createComponentWithSegments({ type });
+        const select = getSegmentsSelect(wrapper);
+
+        assert.equal(select.prop('aria-label'), `Select ${type}`);
+      });
+    });
+
+    [
+      // No selection for groups
+      {
+        segmentsConfig: { type: 'groups' },
+        expectedButtonContent: 'All groups',
+      },
+      // No selection for sections
+      {
+        segmentsConfig: { type: 'sections' },
+        expectedButtonContent: 'All sections',
+      },
+      // 1 known selected item
+      {
+        segmentsConfig: {
+          entries: [
+            {
+              h_authority_provided_id: '1',
+              name: 'Selected Name',
+            },
+          ],
+          selectedIds: ['1'],
+        },
+        expectedButtonContent: 'Selected Name',
+      },
+      // 1 unknown selected group
+      {
+        segmentsConfig: {
+          type: 'groups',
+          entries: [
+            {
+              h_authority_provided_id: '1',
+              name: 'Selected Name',
+            },
+          ],
+          selectedIds: ['3'],
+        },
+        expectedButtonContent: '1 group',
+      },
+      // 1 unknown selected section
+      {
+        segmentsConfig: {
+          type: 'sections',
+          entries: [
+            {
+              h_authority_provided_id: '1',
+              name: 'Selected Name',
+            },
+          ],
+          selectedIds: ['3'],
+        },
+        expectedButtonContent: '1 section',
+      },
+      // Multiple selected groups
+      {
+        segmentsConfig: {
+          type: 'groups',
+          selectedIds: ['1', '3', '8'],
+        },
+        expectedButtonContent: '3 groups',
+      },
+      // Multiple selected sections
+      {
+        segmentsConfig: {
+          type: 'sections',
+          selectedIds: ['1', '3'],
+        },
+        expectedButtonContent: '2 sections',
+      },
+    ].forEach(({ expectedButtonContent, segmentsConfig }) => {
+      it('shows expected segments button content', () => {
+        const wrapper = createComponentWithSegments(segmentsConfig);
+        const content = mount(
+          <div>{getSegmentsSelect(wrapper).prop('buttonContent')}</div>,
+        );
+
+        try {
+          assert.equal(content.text(), expectedButtonContent);
+        } finally {
+          content.unmount();
+        }
+      });
+    });
+
+    [{ entries: [] }, { entries: [{}, {}, {}] }, { entries: [{}] }].forEach(
+      ({ entries }) => {
+        it('shows expected amount of options', () => {
```

Comment:
```suggestion
        it('shows expected number of options', () => {
```

[“Amount” is used for mass nouns, and “number” is used
for count nouns.](https://web.augsburg.edu/english/writinglab/Commonly_Misused_Words.pdf)

Reply:
> “Amount” is used for mass nouns, and “number” is used
> for count nouns.

Thanks!